### PR TITLE
Gitlab example update

### PR DESCRIPTION
**Pull request by naris-silpakit**
*Issue #, if available:*

*Description of changes:*
Updating the GitLab CI/CD example.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/ci_cd/gitlab-ci-cd/README.md [hunk 1]
```diff
@@ -1,29 +1,45 @@
...
+
+    - Make sure the GitLab group that you're referencing in `data.gitlab_group.gitops-eks` exists and that you have access to it.
+
+- ### 400 Error when creating resource
+
+    - If the error contains `{message: {environment_scope: [cannot add duplicated environment scope]}}`, it is likely that an existing Kubernetes integration with the same environment scope was not removed. Remove any Kubernetes clusters with the same environment scope from the GitLab group before redeploying.
+
+- ### What's gitlab-terraform?
+
+    - `gitlab-terraform` is a thin wrapper around the `terraform` binary as part of the [GitLab Terraform docker image](registry.gitlab.com/gitlab-org/terraform-images/stable:latest) used in `.gitlab-ci.yml`.
```
Comment (Zvikan) [inline examples/ci_cd/gitlab-ci-cd/README.md:42]: `registry.gitlab.com/gitlab-org/terraform-images/stable:latest` is image link, probably wanted to have doc link here?

File: examples/ci_cd/gitlab-ci-cd/gitlab-provider.tf [hunk 2]
```diff
@@ -1,23 +1,16 @@
...
-  full_path = "ssp-amazon-eks-terraform-group"
-}
 
-data "gitlab_projects" "ssp-amazon-eks" {
-  group_id          = data.gitlab_group.gitops-eks.id
-  simple            = true
-  search            = "ssp-amazon-eks"
-  include_subgroups = true
+# data source for an existing GitLab group
+data "gitlab_group" "gitops-eks" {
```
Comment (Zvikan) [inline examples/ci_cd/gitlab-ci-cd/gitlab-provider.tf:3]: consider maybe removing this data resource like you did for the project id?

File: examples/ci_cd/gitlab-ci-cd/input.tfvars [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# gitlab-provider.tf
+group_id          = 12345
```
Comment (Zvikan) [inline examples/ci_cd/gitlab-ci-cd/input.tfvars:2]: TF format

File: examples/ci_cd/gitlab-ci-cd/main.tf [hunk 4]
```diff
@@ -61,29 +61,19 @@ data "aws_eks_cluster_auth" "cluster" {
...
-  cluster_name = join("-", [local.tenant, local.environment, local.zone, "eks"])
-
-  terraform_version = "Terraform v1.0.1"
+  vpc_name     = join("-", [var.tenant, var.environment, var.zone, "vpc"])
+  cluster_name = join("-", [var.tenant, var.environment, var.zone, "eks"])
 }
 
 module "aws_vpc" {
-  source  = "terraform-aws-modules/vpc/aws"
-  version = "v3.2.0"
```
Comment (Zvikan) [inline examples/ci_cd/gitlab-ci-cd/main.tf:79]: we should probably keep use a specific version of the module.

File: examples/ci_cd/gitlab-ci-cd/main.tf [hunk 5]
```diff
@@ -61,29 +61,19 @@ data "aws_eks_cluster_auth" "cluster" {
...
+  source = "terraform-aws-modules/vpc/aws"
 
   name = local.vpc_name
-  cidr = local.vpc_cidr
+  cidr = var.vpc_cidr
   azs  = data.aws_availability_zones.available.names
 
-  public_subnets  = [for k, v in slice(data.aws_availability_zones.available.names, 0, 3) : cidrsubnet(local.vpc_cidr, 8, k)]
```
Comment (Zvikan) [inline examples/ci_cd/gitlab-ci-cd/main.tf:85]: we should have the slice back to pickup up to 3 AZs, and if a region doesn't have 3AZs, have the same region `us-west-2` -> we can put a comment explaining about the slice

**Review by vara-bonthu — COMMENTED** [review]
Do we need this PR? I could see another PR for gitlab

File: examples/ci_cd/gitlab-ci-cd/main.tf (hunk 5 above)
Comment (jomcy-amzn) [reply examples/ci_cd/gitlab-ci-cd/main.tf:85]: added , Updated code

File: examples/ci_cd/gitlab-ci-cd/input.tfvars (hunk 3 above)
Comment (jomcy-amzn) [reply examples/ci_cd/gitlab-ci-cd/input.tfvars:2]: line removed

File: examples/ci_cd/gitlab-ci-cd/gitlab-provider.tf (hunk 2 above)
Comment (jomcy-amzn) [reply examples/ci_cd/gitlab-ci-cd/gitlab-provider.tf:3]: File deleted

File: examples/ci_cd/gitlab-ci-cd/README.md (hunk 1 above)
Comment (jomcy-amzn) [reply examples/ci_cd/gitlab-ci-cd/README.md:42]: link updated

**Comment by naris-silpakit** [comment]
Closing PR as it's a duplicate of https://github.com/aws-samples/aws-eks-accelerator-for-terraform/pull/215

**Pull request closed without merging**